### PR TITLE
[inductor] Replace torch.allclose with torch.testing.assert_close in test_fx_fusion

### PR DESCRIPTION
**Pull request by robert-hardwick**
Preventative fix of a test failure with oneDNN v3.5 upgrade where order of float32 arithmetic may change in torch.admm ( bias term can be at the start or end of the arithmetic ) resulting in slightly different output due to float32 precision loss.

Replaced occurrences of torch.allclose with ~~torch._dynamo.testing.same~~  torch.testing.assert_close which is the recommended approach as per this issue https://github.com/pytorch/pytorch/issues/56544 ,the default tolerance is more relaxed than torch.allclose which satisfies the test with upcoming oneDNN change.

This should fix aarch64 ci failures in #129932 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @milpuz01

### CODE REVIEW

**Comment by malfet** [comment]
@robert-hardwick please fix the lint?

File: test/inductor/test_fx_fusion.py [hunk 1]
```diff
@@ -66,7 +67,7 @@ def test_kwarg3(x, y):
         ]
         for f in [test_kwarg, test_arg, test_arg2, test_kwarg2, test_kwarg3]:
             traced = trace_func(f, inputs)
-            self.assertTrue(torch.allclose(f(*inputs), traced(*inputs)))
+            self.assertTrue(same(f(*inputs), traced(*inputs)))
```
Comment (malfet) [inline test/inductor/test_fx_fusion.py:70]: Why not do something like that? And if lower tolerance is specific to arm platform, why not condition it on `platform.processor() == 'aarch64'`?
```suggestion
            self.assertEqual(f(*inputs), traced(*inputs), rtol=1e-4, atol=1e-4)
```

Comment (robert-hardwick) [reply test/inductor/test_fx_fusion.py]: I have fixed the lint and improved this by using `[torch.testing.assert_close](https://pytorch.org/docs/stable/testing.html#torch.testing.assert_close)` which seems to be the recommended approach. The default tolerances appears to satisfy all hardware types so no need to put a platform condition in.

**Comment by yanbing-j** [comment]
Hi @malfet , could you please help review this PR? Thanks!

**Review by jgong5 — APPROVED** [review]

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**